bundles/scripting/sightly/engine/src/main/java/org/apache/sling/scripting/sightly/impl/engine/extension/ResourceRuntimeExtension.java

```diff
@@ -229,16 +238,24 @@ private Object getAndRemoveOption(Map<String, Object> options, String property)
 
     private void includeResource(final Bindings bindings, PrintWriter out, String script, String dispatcherOptions, String resourceType) {
         if (StringUtils.isEmpty(script)) {
-            LOG.error("Script path cannot be empty");
+            LOG.error("Resource path cannot be empty");
         } else {
-            SlingHttpServletResponse customResponse = new PrintWriterResponseWrapper(out, BindingsUtils.getResponse(bindings));
             SlingHttpServletRequest request = BindingsUtils.getRequest(bindings);
             script = normalizePath(request, script);
-
             Resource includeRes = request.getResourceResolver().resolve(script);
             if (includeRes instanceof NonExistingResource || includeRes.isResourceType(Resource.RESOURCE_TYPE_NON_EXISTING)) {
                 includeRes = new SyntheticResource(request.getResourceResolver(), script, resourceType);
             }
+            includeResource(bindings, out, includeRes, dispatcherOptions, resourceType);
+        }
+    }
+
+    private void includeResource(final Bindings bindings, PrintWriter out, Resource includeRes, String dispatcherOptions, String resourceType) {
+        if (includeRes == null) {
+            LOG.error("Resource cannot be null");
```

Comment:
I would prefer throwing an exception here as well. That way programming mistakes are detected earlier.